genie-common/src/main/java/com/netflix/genie/common/util/AdaptiveBackOffAdaptor.java

```diff
@@ -0,0 +1,55 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.common.util;
+
+/**
+ * Wrapper for Runnable that artificially delays the task execution.
```

Comment:
Here you call it a "wrapper" but you named the class `Adapter`. To me it seems like a wrapper as it's not really adapting anything

Reply:
At least GoF canon, Adapter and Wrapper are the same thing https://en.wikipedia.org/wiki/Adapter_pattern

Reply:
(and it's adapting a Runnable scheduled with a random backoff to look like one that is scheduled at fixed rate)